### PR TITLE
storage/seg: add clear function

### PR DESCRIPTION
**Pull request by brayniac**
Adds a function to clear the stored data.

### CODE REVIEW

File: src/rust/storage/seg/src/ttl_buckets/ttl_bucket.rs [hunk 1]
```diff
@@ -106,6 +106,34 @@ impl TtlBucket {
         }
     }
 
+    // clear segments from this TtlBucket, returns the number of segments expired
+    pub(super) fn clear(&mut self, hashtable: &mut HashTable, segments: &mut Segments) -> usize {
```
Comment (thinkingfish) [inline src/rust/storage/seg/src/ttl_buckets/ttl_bucket.rs:110]: I personally would prefer `bulk_clear` over `clear` for all the new `clear` traits, and keep the original `segment.clear` as the only `clear` function.

File: src/rust/storage/seg/src/ttl_buckets/ttl_buckets.rs [hunk 2]
```diff
@@ -125,6 +125,19 @@ impl TtlBuckets {
...
         expired
     }
+
+    pub(crate) fn clear(&mut self, hashtable: &mut HashTable, segments: &mut Segments) -> usize {
+        let start = Instant::now();
+        let mut cleared = 0;
+        for bucket in self.buckets.iter_mut() {
+            cleared += bucket.clear(hashtable, segments);
+        }
+        segments.set_flush_at(CoarseInstant::now());
```
Comment (thinkingfish) [inline src/rust/storage/seg/src/ttl_buckets/ttl_buckets.rs:135]: Considering the semantics of the memcached `flush_all` command, which allows delayed flushing, I wonder if we should set this outside this function. This way, we can set that `flush_at` in segments when we receive the command, and let the storage module handle the actual flushing internally (later).

File: src/rust/storage/seg/src/ttl_buckets/ttl_bucket.rs (hunk 1 above)
Comment (brayniac) [reply src/rust/storage/seg/src/ttl_buckets/ttl_bucket.rs:110]: Happy to make the change if you prefer. I was trying to keep the high-level API closer to the std library HashMap: https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.clear

Comment (thinkingfish) [reply src/rust/storage/seg/src/ttl_buckets/ttl_bucket.rs:118]: Ah I see. this is fine

**Review by thinkingfish — APPROVED** [review]

**Pull request merged**